mongo/integration/mtest/setup.go

```diff
@@ -60,7 +60,8 @@ func setupClient(cs connstring.ConnString, opts *options.ClientOptions) (*mongo.
 	if opts.ServerAPIOptions == nil && testContext.requireAPIVersion {
 		opts.SetServerAPIOptions(options.ServerAPI(driver.TestServerAPIVersion))
 	}
-	// for sharded clusters, pin to one host
+	// for sharded clusters, pin to one host. Due to how the cache is implemented on 4.0 and 4.2, behavior
+	// can be inconsistent when multiple mongoses are used
```

Comment:
Ah, I think I missed this in the prior PR. Nit: Is there a server ticket to link to and include in the comment?

Reply:
Not one that I know. But considering that it got fixed but not backported in 4.4 I don't think it really matters